### PR TITLE
fix: handle chunked prefill with speculative decoding

### PR DESCRIPTION
**Pull request by junstar92**
### 🚀 Summary of Changes

Fixes speculative decoding failures during intermediate chunked prefill steps.
This PR updates the RBLN V1 model runner and EAGLE proposer so that speculative decoding handles chunked prefill correctly:

- Detects intermediate chunked prefill steps where no logits are sampled.
- Avoids generating draft tokens from empty logits during intermediate prefill.
- For Medusa, returns empty draft token lists for active requests.
- For EAGLE/EAGLE3, runs a prefill-only draft model forward to populate draft KV cache without producing draft tokens.
- Keeps final prefill and decode behavior unchanged.
- Adds unit coverage for Medusa, EAGLE, and EAGLE prefill-only paths.

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [x] 🧬 Core engine changes (`core`)
* [x] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run unit tests:

```
pytest \
  tests/torch_compile/unit/v1/worker/test_rbln_model_runner.py::TestProposeDraftTokenIdsIntermediatePrefill \
  tests/torch_compile/unit/v1/spec_decode/test_eagle.py::test_prefill_only_keeps_unpadded_batch_shape_and_returns_none
```

2. Run manual Medusa/EAGLE chunked prefill tests.
Update the prompt in `examples/experimental/run_specdec_medusa_test.py` and `examples/experimental/run_specdec_eagle_test.py` so that the prompt length is longer than the configured `max_num_batched_tokens`, then run the scripts.

---

### 📸 Screenshots / Logs (if applicable)

N/A

---

### 📋 Checklist

* [x] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [x] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

**Comment by junstar92** [comment]
@rebel-jinhwan @rebel-jiwoopark 
This PR depends on the EAGLE changes for the updated attention inference in #546. Please merge #546 first before this PR.

**Review by rebel-jinhwan — APPROVED** [review]
LGTM

File: vllm_rbln/v1/spec_decode/eagle.py [hunk 1]
```diff
@@ -336,6 +336,122 @@ def propose(
...
+            num_padded_tokens=num_padded_tokens,
+            additional_kwargs=build_kv_cache_forward_context_kwargs(
+                getattr(self.runner, "kv_cache_bases", None)
+            ),
+        ):
+            _, _ = self.model_executable(
+                input_ids=input_ids,
+                positions=positions,
+                hidden_states=hidden_states,
+                inputs_embeds=inputs_embeds,
```
[the remark this answers is not in the document]
Comment (junstar92) [reply vllm_rbln/v1/spec_decode/eagle.py:454]: Multimodel in Eagle/Eagle3 is not supported yet.
I added an error raise for multimodal inputs, and I'll revisit it when supporting multimodal.
